doc/aaa/ldap.md

```diff
@@ -0,0 +1,2617 @@
+     |           |  +--rw oc-aaa-ldap-ext:source-address?               oc-inet:ip-address
+     |           |  +--rw oc-aaa-ldap-ext:vrf-name?                     string
+     |           |  +--rw oc-aaa-ldap-ext:idle-time-limit?              uint32
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-group?               string
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-shadow?              string
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-netgroup?            string
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-sudoers?             string
+     |           |  +--rw oc-aaa-ldap-ext:nss-initgroups-ignoreusers?   string
+     |           |  +--rw oc-aaa-ldap-ext:pam-filter?                   string
+     |           |  +--rw oc-aaa-ldap-ext:pam-login-attribute?          string
+     |           |  +--rw oc-aaa-ldap-ext:pam-group-dn?                 string
+     |           |  +--rw oc-aaa-ldap-ext:pam-member-attribute?         string
+     |           |  +--rw oc-aaa-ldap-ext:sudoers-base?                 string
+     |           |  +--rw oc-aaa-ldap-ext:scope?                        enumeration
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-passwd?              string
+     |           +--ro oc-aaa-ldap-ext:state
+     |           +--rw oc-aaa-ldap-ext:maps
+     |              +--rw oc-aaa-ldap-ext:map* [name from]
+     |                 +--rw oc-aaa-ldap-ext:name      -> ../config/name
+     |                 +--rw oc-aaa-ldap-ext:from      -> ../config/from
+     |                 +--rw oc-aaa-ldap-ext:config
+     |                 |  +--rw oc-aaa-ldap-ext:name?   enumeration
+     |                 |  +--rw oc-aaa-ldap-ext:from?   string
+     |                 |  +--rw oc-aaa-ldap-ext:to?     string
+     |                 +--ro oc-aaa-ldap-ext:state
+     |                    +--ro oc-aaa-ldap-ext:name?   enumeration
+     |                    +--ro oc-aaa-ldap-ext:from?   string
+     |                    +--ro oc-aaa-ldap-ext:to?     string
+...
+...
+
+
+```
+
+## REST/gNMI
+
+REST, and gNMI support would be needed by writing transformer annotation,
+overloads, and SONiC Yang.
+
+## KLISH
+
+KLISH commands (XML, actioners, renderers) needs to be written for
+
+- AAA authentication ldap option.
+
+```
+sonic(config)# [no] aaa authentication login-method { ldap | local }*
+```
+
+- AAA authorization login ldap option..
+
+```
+sonic(config)# [no] aaa authorization login {ldap|local}
+```
+
+- AAA name-service.
+
+```
+sonic(config)# [no] aaa name-service passwd { ldap|login|local }
+sonic(config)# [no] aaa name-service shadow { ldap|login|local }
+sonic(config)# [no] aaa name-service group { ldap|login|local }
+sonic(config)# [no] aaa name-service netgroup { ldap|local }
+sonic(config)# [no] aaa name-service sudoers { ldap|local }
+```
+
+- LDAP global|nss|pam|sudo server authentication, authorization, name-service parameters.
+
+
+```
+ldap-server timelimit  <0 - 65535>
```

Comment:
TL;DR; 
It would be interesting to see how Cisco configures LDAP. Many customers are Cisco-trained and are used to do things the Cisco way. Using CLI commands that are identical (or similar) to Cisco's in SONiC would most likely improve customer experience.